### PR TITLE
Primitives Proto type to Json schema

### PR DESCRIPTION
**Pull request by lideke**
Add ProtoToJsonSchema primitives type converter
Add GetJsonSchema Service

### CODE REVIEW

File: core/src/main/java/io/anemos/metastore/core/proto/ProtoToJsonSchema.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      case FIXED32:
+      case SFIXED32:
+        return "integer";
+      case FLOAT:
+      case DOUBLE:
+        return "numeric";
+      case BOOL:
+        return "boolean";
+      case STRING:
+      case BYTES:
```
Comment (alexvanboxel) [inline core/src/main/java/io/anemos/metastore/core/proto/ProtoToJsonSchema.java]: I would expect more retrictions for BYTES, like for example the use of : 
https://json-schema.org/latest/json-schema-validation.html#rfc.section.8.3 `"contentEncoding": "base64"`

File: core/src/main/java/io/anemos/metastore/core/proto/ProtoToJsonSchema.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      case INT64:
+      case SINT64:
+      case FIXED64:
+      case SFIXED64:
+      case INT32:
+      case UINT32:
+      case SINT32:
+      case FIXED32:
+      case SFIXED32:
+        return "integer";
```
Comment (alexvanboxel) [inline core/src/main/java/io/anemos/metastore/core/proto/ProtoToJsonSchema.java]: try to look deeper into the spec for getting everything more restricted.

File: core/src/main/java/io/anemos/metastore/core/proto/ProtoToJsonSchema.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            .withDescription(fieldDescriptor.toString() + " is not implemented yet.")
+            .asRuntimeException();
+    }
+  }
+
+  private ObjectNode toRecord(Descriptors.Descriptor descriptor) {
+    ObjectMapper mapper = new ObjectMapper();
+
+    final ObjectNode node = mapper.createObjectNode();
+    node.put("title", descriptor.getName());
```
Comment (alexvanboxel) [inline core/src/main/java/io/anemos/metastore/core/proto/ProtoToJsonSchema.java]: Probably we need the full name, not only the message name to make it unique,

File: core/src/main/java/io/anemos/metastore/core/proto/ProtoToJsonSchema.java (hunk 1 above)
Comment (lideke) [reply core/src/main/java/io/anemos/metastore/core/proto/ProtoToJsonSchema.java]: Added restrictive contentEncoding

File: core/src/main/java/io/anemos/metastore/core/proto/ProtoToJsonSchema.java (hunk 2 above)
Comment (lideke) [reply core/src/main/java/io/anemos/metastore/core/proto/ProtoToJsonSchema.java]: added minimum value for int32 and int64 to 0

Comment (lideke) [reply core/src/main/java/io/anemos/metastore/core/proto/ProtoToJsonSchema.java]: also on uint32, uint64, fixed32 and fixed64

File: core/src/main/java/io/anemos/metastore/core/proto/ProtoToJsonSchema.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+            .asRuntimeException();
+    }
+  }
+
+  private ObjectNode getJsonRestrictives(ObjectMapper mapper,Descriptors.FieldDescriptor.Type fieldDescriptor) {
+    final ObjectNode nodeRestrict = mapper.createObjectNode();
+
+    switch (fieldDescriptor) {
+      case INT32:
+      case INT64:
```
Comment (alexvanboxel) [inline core/src/main/java/io/anemos/metastore/core/proto/ProtoToJsonSchema.java]: INT32/64 are signed ( https://developers.google.com/protocol-buffers/docs/proto#scalar )

File: core/src/main/java/io/anemos/metastore/core/proto/ProtoToJsonSchema.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+      case UINT32:
+      case UINT64:
+      case FIXED32:
+      case FIXED64:
+        nodeRestrict.put("minimum", 0);
```
Comment (alexvanboxel) [inline core/src/main/java/io/anemos/metastore/core/proto/ProtoToJsonSchema.java]: what about the max value?

Comment (lideke) [reply core/src/main/java/io/anemos/metastore/core/proto/ProtoToJsonSchema.java]: add more restrictives rules

**Pull request merged**